bin/migrate_acl_authz.py

```diff
@@ -0,0 +1,279 @@
+the record previously had both a program and a project, since the authz field requires
+access to all the listed items, only the project should end up in `authz` (since
+requiring the program would omit access to users who can access only the project).
+
+Furthermore, there are two ways to represent the arborist resources that go into
+`authz`: the path (human-readable string) and the tag (random string, pseudo-UUID). The
+tags are what we want to ultimately put into the `authz` field, since these are
+persistent whereas the path could change if resources are renamed.
+"""
+
+import argparse
+import os
+import re
+import sys
+
+from cdislogging import get_logger
+import requests
+import sqlalchemy
+from sqlalchemy import and_, func
+from sqlalchemy.engine import create_engine
+from sqlalchemy.exc import OperationalError
+
+from indexd.index.drivers.alchemy import IndexRecord, IndexRecordAuthz
+
+
+logger = get_logger("migrate_acl_authz")
+
+
+def main():
+    args = parse_args()
+    sys.path.append(args.path)
+    try:
+        from local_settings import settings
+    except ImportError:
+        logger.info("Can't import local_settings, import from default")
+        from indexd.default_settings import settings
+    driver = settings["config"]["INDEX"]["driver"]
+    try:
+        acl_converter = ACLConverter(args.arborist, getattr(args, "sheepdog"))
+    except EnvironmentError:
+        logger.error("can't continue without database connection")
+        sys.exit(1)
+
+    with driver.session as session:
+        q = session.query(IndexRecord)
+        if getattr(args, "start_did"):
+            q = q.filter(IndexRecord.did >= args.start_did)
+        for record in windowed_query(session, q, IndexRecord.did, int(args.chunk_size)):
+            if not record.acl:
+                logger.info(
+                    "record {} has no acl, setting authz to empty"
+                    .format(record.did)
+                )
+                record.authz = []
+                continue
+            try:
+                authz = acl_converter.acl_to_authz(record)
+                if authz:
+                    record.authz = [IndexRecordAuthz(did=record.did, resource=authz)]
+                    logger.info("updated authz for {} to {}".format(record.did, authz))
+                else:
+                    record.authz = []
+                    logger.info("updated authz for {} to empty list".format(record.did))
+                session.add(record)
+            except EnvironmentError as e:
+                msg = "problem adding authz for record {}: {}".format(record.did, e)
+                logger.error(msg)
+    logger.info("finished migrating")
+    return
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--path", default="/var/www/indexd/", help="path to find local_settings.py",
+    )
+    parser.add_argument(
+        "--sheepdog-db", dest="sheepdog", help="URI for the sheepdog database"
```

Comment:
you can default this to `True` instead of doing a getattr() later, either way works

Reply:
oh wait, I think I misunderstood this. it's the db connection string :man_facepalming: nix that then